zerver/decorator.py

```diff
@@ -1062,3 +1062,23 @@ def _wrapped_view_func(
         return response
 
     return _wrapped_view_func
+
+
+def slack_error_handler(view_func: Callable[..., HttpResponse]) -> Callable[..., HttpResponse]:
+    """
+    A decorator that catches JsonableError exceptions and returns a
+    Slack-compatible error response in the format:
+    {ok: false, error: "error message"}.
+    """
+
+    @wraps(view_func)
+    def wrapped_view(
+        request: HttpRequest, *args: ParamT.args, **kwargs: ParamT.kwargs
+    ) -> HttpResponse:
+        try:
+            return view_func(request, *args, **kwargs)
+        except JsonableError as error:
+            error_message = error.msg_format().format(**getattr(error, "data", {}))
+            return JsonResponse({"ok": False, "error": error_message}, status=400)
+
+    return wrapped_view
```

Comment:
Let's keep this code in the Slack webhook's `view.py` file; just because it's a decorator doesn't mean it's an auth decorator that belongs here.